extensions/shared/register-jetpack-block.js

```diff
@@ -44,9 +57,23 @@ export default function registerJetpackBlock( name, settings, childBlocks = [] )
 		return false;
 	}
 
+	let blockTitle = settings.title;
+	const blockTags = [];
```

Comment:
Minor: it might seem obvious when you've been working on it for a while, but I wonder whether it could be helpful to add a comment here detailing what we're doing here. It was immediately obvious to me even though I had some context. For others it could be even less clear.

For example:

> Some Blocks should be "tagged" to highlight various attributes such as being part of a paid plan or being in a beta programme. We add these to the Block's visible `title` in order that they show up in the picker.

Alternatively, perhaps it would be better to handle adding "tags" in a dedicated method. This might remove the need to document with a comment as the code would be self describing. Perhaps:

`buildBlockTags` and `buildBlockTitle`?

Reply:
I've refactored the implementation here: https://github.com/Automattic/jetpack/pull/14739/commits/25129a6679a36b5f715f9fd7eeb86994d164e1cc

This adds `buildBlockTags` and `buildBlockTitle` functions and builds the title with `buildBlockTitle( settings.title, buildBlockTags( name, requiredPlan ) )`